### DISCUSSION TITLE
Support for filepath() Function to Retrieve Source File Path as a Column During Ingestion

### DISCUSSION THREAD

**Discussion by rohithreddykota** [duckdb/duckdb#14791] [Ideas]
### Description

It would be highly beneficial if DuckDB could natively support a filename() or filepath() function to retrieve the full path of source files as a column when ingesting data from external storage systems (e.g., S3, local file systems). This feature would streamline data ingestion workflows by automatically including the file path of each row’s source file as a column in SQL queries, especially when reading from multiple files.

### Use Case
In many data engineering workflows, it’s crucial to know the origin of each row, especially when working with partitioned data stored in multiple files. Currently, DuckDB users need to use external scripting to add the file path as a column, which requires additional overhead and complexity. A filepath() function in SQL would make DuckDB more efficient and easier to use for such cases.

**Example Syntax**
The feature could look something like this:

```sql
SELECT *, filepath() AS source_file
FROM read_parquet('s3://your-bucket-name/path/to/files/*.parquet');
```

In this example, filepath() would return the full path of each file being read, effectively adding a source_file column to the result with the path of each respective row’s source file.

**Comment by Tishj** [comment]
See <https://duckdb.org/docs/data/csv/overview.html#parameters> `filename`

If memory serves this option also works for read_parquet
